src/mongo.ts

```diff
@@ -0,0 +1,137 @@
+import { Db, MongoClient } from "mongodb";
+
+import { config } from "./config";
+import { ReplSetConfig, ReplSetStatus } from "./types";
+import { log } from "./log";
+import { range, sleep } from "./utils";
+
+const getDb = async (host: string = "127.0.0.1"): Promise<Db> => {
+  const mongoConfig = config.mongo;
+  const authConfig = mongoConfig.auth;
+
+  const uri =
+    authConfig ?
+      `mongodb://${encodeURIComponent(authConfig.username)}:${encodeURIComponent(authConfig.password)}@${host}:${mongoConfig.port}`
```

Comment:
I see you removed the `/{database}` - any particular reason? Wasn't it useful/needed for when authenticating user is created outside the `admin` db?

Reply:
Or is the `.db(mongoConfig.database)` actually when the connection & auth happens, hence you are providing the db there?